src/odemis/driver/test/npmc_test.py

```diff
@@ -235,6 +228,16 @@ def test_multimove(self):
         f2.result()
         test.assert_pos_almost_equal(orig_pos, self.dev.position.value, **COMP_ARGS)
 
+    def test_reference(self):
+        """
+        Test referencing
+        """
+        self.dev.moveAbs({'x': 0.3}).result()
+        self.dev.reference(('x',)).result()
+        # The new position shoudl be the origin (0)
+        origin = {'x': 0}
```

Comment:
Please also test cancelling a `reference()`. You can move "far away" from the origin, and then start the referencing, wait a tiny bit (eg, 0.01s) to make sure it's started, and stop the referencing. Check that .referenced is False, and that the position is not 0.